internal/component/loki/relabel/relabel_test.go

```diff
@@ -220,7 +224,10 @@ func TestCache(t *testing.T) {
 	e.Labels = lsets[2]
 	c.receiver.Chan() <- e
 
-	time.Sleep(100 * time.Millisecond)
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		require.EqualValues(t, 3, receivedMessages.Load())
+	}, 250*time.Millisecond, 25*time.Millisecond)
```

Comment:
Some GC pauses or other hangs in CI can take multiple seconds, so I would always go with at least 3 seconds limit.